### DISCUSSION TITLE
Client-key header checked by the Api Gateway even though the endpoint does not require an API key

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#335] [Product Feedback]
### Which section?

Console

### Which is the version?

11.7.0

### Description

Hi!

I have noticed that, by default, the API Gateway (both Envoy and Nginx) checks the Client-Key header even though the endpoint I am calling does not require an API key. In particular, in case of an endpoint that does not require an API key:
- if the client sends an API Key that is not listed in the project -> 401 Unauthorized
- if the client sends an API Key that is listed in the project -> 200 OK
- if the client sends an empty Client-Key header -> 200 OK with Nginx, 403 Forbidden with Envoy

Is this the intended behaviour?

I would like the API Gateway to ignore the Client-Key header if the endpoint does not require an API key. This is because maybe that header will be checked in a downstream Mia-Platform project.

Thanks in advance!

**Comment by fredmaggiowski** [comment]
~To be required means that it is not required, not that you can send anything to the relative header~ 😄

EDIT: let me rephrase (lol)
when you disable the require api key flag you just allow for requests without the key, if something is provided in the header it is checked

If you send it, the API key is still evaluated to understand what type of client is performing the HTTP request.

There might be an additional feature to completely skip the header validation but I honestly don't find it that helpful as it may lead to security issues when dealing with keys you do not want to serve anymore.
